packages/fetch-engine/src/__tests__/download.test.ts

```diff
@@ -450,20 +531,34 @@ Array [
         'darwin-arm64',
         'debian-openssl-1.0.x',
         'debian-openssl-1.1.x',
+        'debian-openssl-3.0.x',
         'linux-arm64-openssl-1.0.x',
         'linux-arm64-openssl-1.1.x',
+        'linux-arm64-openssl-3.0.x',
         'rhel-openssl-1.0.x',
         'rhel-openssl-1.1.x',
+        'rhel-openssl-3.0.x',
         'windows',
         'linux-musl',
       ],
       version: FIXED_BINARIES_HASH,
     })
     const after2 = Date.now()
-    // value on Mac: 33ms
-    // value on GH Actions: ?
-    // https://github.com/prisma/prisma/runs/1176632754
-    const took2 = after2 - before2
-    expect(took2).toBeLessThan(10000)
+    const timeInMsToDownloadAllFromCache2 = after2 - before2
+    console.debug(
+      `3 - With cache2: Engines were already present
+It took ${timeInMsToDownloadAllFromCache2}ms to execute download() for all binaryTargets.`,
+    )
+
+    // This is a rather high number to avoid flakiness in CI
+    expect(timeInMsToDownloadAll).toBeLessThan(20_000)
+    expect(timeInMsToDownloadAllFromCache1).toBeLessThan(20_000)
+    expect(timeInMsToDownloadAllFromCache2).toBeLessThan(20_000)
+
+    // Using cache should be faster
+    expect(timeInMsToDownloadAllFromCache1).toBeLessThan(timeInMsToDownloadAll)
+    expect(timeInMsToDownloadAllFromCache2).toBeLessThan(
+      timeInMsToDownloadAllFromCache1,
+    )
```

Comment:
I feel like this might be flaky. It should ideally be true, but I don't see how we can test this reliably. On a dedicated machine where nothing else is running — it might work, but probably not always on CI.

Reply:
Let's try and see? If it's flaky we change it